### ISSUE TITLE
Commit to a branch naming convention.

### ISSUE THREAD

**Issue by BenGitsCode** [BenGitsCode/today-i-learned#12]
Torn between creating branches for each member (i.e. github user name ) and each contributor uses that for pushing TIL's

OR 

branch for each TIL, waxed after merge

OR 

branches per directory with a need to be mindful of other collaborator's changes.

**Comment by laurpaik-zz** [comment]
If we have a branch per directory can't you just delete it once the pull request is merged? Or is that frowned upon?

**Comment by BenGitsCode** [comment]
That's precisely why I opened the issue, I'm not sure.  That might be fine but while I think git would protect and conflicting references, I'm not positive, and I think the former two may be safer. 

I think if you're a regular contributor like you or me, a branch per user makes sense. It would hold all your TILs. But some convention adherence in my head doesn't like creating more isolated branches. I'm not sure if it matters but I at least want this discussion going. Even if we don't make any decision right away, I want this issue to close with a better answer than I have right now.

**Comment by laurpaik-zz** [comment]
yea idk I've just been naming them after the directories, but I also haven't been using the same ones you've been using because I've been job hunting more than experimenting with apps. I guess if I really wanted to, I could make a `laurpaik-development` branch and off of that make whatever local branch name I want, merge it to the `laurpaik-development` branch, and that one's the one I'll push and use for pull requests. Writing it all out makes it sound like a hassle though.

**Comment by BenGitsCode** [comment]
I think so far what you're saying to me right now (next to me IRL not in that comment)

Branches named after directories merged locally into a branch named after user <githubUserName> and then PRs made to this repository 👍

**Comment by BenGitsCode** [comment]
Naming this is hard.

**Comment by laurpaik-zz** [comment]
I mean... I'm all for the username branches. I'd rather do that than have to coordinate with other users about what to do when you both learn something about the same app.

**Comment by laurpaik-zz** [comment]
I have an `alexa` branch checked out of `laurpaik`... I merged that to `laurpaik` locally when I was done, and then I pushed `laurpaik` and submitted a pull request from there.

**Comment by BenGitsCode** [comment]
That's what I've ben doing, and I think I do like that best. I'll open an issue to outline this in the `CONTRIBUTE.MD`

**Comment by laurpaik-zz** [comment]
If we're doing this one-branch-per-user thing, are we okay with having pull requests do multiple things?
Reason for asking: Usually I would have submitted two pull requests just now, one for the `logic trivia`, and one for the `about-me stuff`, but because I just have the one branch I can't. If you're okay with this I'm cool with it too, and if not... well it's too late this time, but can fix for next time

**Comment by BenGitsCode** [comment]
Yeah I think it makes more sense to make and delete branches per PR/TIL. Technically maybe we should be merging them to development and _then_ master. But since TILs don't depend on each other or actually run at all, it's okay for now. 

So far it seems like branching per TIL makes the most sense and avoids confusion.

**Comment by laurpaik-zz** [comment]
👍 Does this mean we can close this, or did you want to keep it open in case something changes?

**Comment by BenGitsCode** [comment]
I'll label it for now and icebox it, but I don't think it's definitive enough to close. 

I encourage any and all to add to this discussion until we find a satisfactory workflow.
